api/server/server.go

```diff
@@ -787,15 +789,19 @@ func (s *Server) postImagesCreate(eng *engine.Engine, version version.Version, w
 		}
 		imageImportConfig.ContainerConfig = newConfig
 
-		opErr = s.daemon.Repositories().Import(src, repo, tag, imageImportConfig)
-	}
+		err = s.daemon.Repositories().Import(src, repo, tag, imageImportConfig)
 
-	if opErr != nil {
+	}
+	if err != nil {
+		if !output.Flushed() {
+			return err
+		}
 		sf := streamformatter.NewStreamFormatter(useJSON)
-		return fmt.Errorf(string(sf.FormatError(opErr)))
+		output.Write(sf.FormatError(err))
```

Comment:
output.Flushed() is used to check if we already wrote on the socket to my best knowledge. If we did, we cannot return an error (otherwise we will have duplicate write headers and such things)
See also`postImagesPush`.